crescent/plugin.py

```diff
@@ -0,0 +1,41 @@
+from __future__ import annotations
+
+import importlib
+from typing import TypeVar, TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from .bot import Bot
+    from .internal.meta_struct import MetaStruct
+
+_T = TypeVar("_T", bound="MetaStruct")
+
+
+class Plugin:
+    def __init__(self, name: str) -> None:
+        self.name = name
+        self._children: list[MetaStruct] = []
+
+    def include(self, obj: _T) -> _T:
+        self._children.append(obj)
+        return obj
+
+    def setup(self, bot: Bot) -> None:
```

Comment:
Should `setup` be public? The user shouldn't touch this so I think it should be protected.